metagoofil.py

```diff
@@ -238,6 +236,16 @@ def _split_lines(self, text, width):
         return argparse.HelpFormatter._split_lines(self, text, width)
 
 
+def pos(type):
+    def check(value):
```

Comment:
I've never seen this function convention before.  It was hard to google for as well...is this called something specifically?

How is the actual `value` from argparse being passed?  Are you overriding a `.check()` function within argparse or something?

I definitely like the spirit of this and want to keep it, but anyway it could be simplified?  Or do you have some documentation recommending this for Python argparse type checking?

Reply:
The `type` argument of argparse's `add_argument()` is a callable that takes a string, converts it to the desired type, and can do some basic type-checking (see [https://docs.python.org/3/library/argparse.html#type](https://docs.python.org/3/library/argparse.html#type)). 

Here `pos(T)` returns such a callable (called `check()`, but the name has no special meaning) that maps a string to a result of type `T`. It might be simpler to define separate `pos_float()` and `pos_int()`, and then use `type=pos_int` instead of `type=int`:
```python
def pos_int(value):
    try:
        value_int = int(value)
        assert value_int >= 0
        return value_int
    except (AssertionError, ValueError):
        raise argparse.ArgumentTypeError(f"invalid value '{value}', must be an int >= 0")
```

In this particular case, I think checking that these 3 args are positive after parsing is fine, a callable returning a callable / using custom "types" is a bit overkill for such a simple interface. 
The improvement provided here is minimal, this kind of custom type is more useful and relevant for a larger number of arguments with similar constraints, which isn't really the case here.

Reply:
Learn something new every day!  If you could make it two separate functions:

* `positive_int()`
* `positive_float()`

and keep the simple function check you have above instead of a callable returning a callable...that'd be perfect.